pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -348,6 +348,11 @@ func (plugin *kubenetNetworkPlugin) setup(namespace string, name string, id kube
 	if err != nil {
 		return err
 	}
+
+	// This also add the current pod to the running pods list because for rkt, when we set up the network,
+	// the pod is not currently running.
+	runningPods = append(runningPods, &hostport.RunningPod{Pod: pod, IP: ip4})
```

Comment:
I was going to suggest that perhaps the new pod's IP should be passed into OpenPodHostportsAndSync() and then that function could handle adding the current pod to runningPods if it wasn't there already.